server/src/main/kotlin/org/javacs/kt/CompilerClassPath.kt

```diff
@@ -95,6 +98,10 @@ class CompilerClassPath(private val config: CompilerConfiguration) : Closeable {
         workspaceRoots.add(root)
         javaSourcePath.addAll(findJavaSourceFiles(root))
 
+        val outputDir = Paths.get(root.toString(), "kls").toFile()
```

Comment:
I would prefix that directory with a dot, e.g. `.kls`, also perhaps a less generic name? Alternatively, could we perhaps use the `storagePath` introduced in #337, so we don't have to put custom build outputs into the user's project?

Reply:
Your second suggestion would be perfect, but unfortunately it'll be tricky to implement (although doable).

The problem is that the JDT LS extension needs to know where to find the build outputs. A directory inside the user's project is an easy solution, because the JDT LS extension can know about this without any help. If we use a different directory, we will need to establish communication between the JDT LS extension and the kotlin language server. This is obviously doable and ideal, but it is trickier to implement.

I would suggest using `.kls` for now (your first suggestion) and in a later PR we could implement the other approach. What do you think?

Reply:
Hey. So I thought a little more about this and I think I might have an alternative approach to this that would make things a lot cleaner.

Instead of using custom project importers on the JDT LS extension, we could have a [delegateCommandHandler](https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/plugin.xml#L4) that adds the build output location to the java project classpath. With this approach, the java project would be imported using the normal JDT project importers but we would change the classpath as soon as the KLS is setup. In order for this to work, the KLS would need to be able to call the `delegateCommandHandler` once it knows where it will store the build outputs. This call would probably need to go through the vscode extensions, since a direct communication between the language servers is tricky. The cool thing about this approach is that the JDT LS would just work normally at first and as soon as the KLS starts, it would be notified to update its project. This approach would also allow us to use temp directories as a default build output location, which is a lot cleaner (especially for people who are not even working on a Java + Kotlin project)